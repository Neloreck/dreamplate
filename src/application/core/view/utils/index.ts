export * from "@/core/view/utils/rendering";
