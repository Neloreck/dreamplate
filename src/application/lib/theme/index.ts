export * from "@/lib/theme/types";
export * from "@/lib/theme/utils";
