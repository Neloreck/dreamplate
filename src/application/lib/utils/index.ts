export * from "@/lib/utils/local_storage";
export * from "@/lib/utils/delay";
