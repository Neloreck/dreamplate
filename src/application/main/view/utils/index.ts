export * from "@Main/view/utils/rendering";
