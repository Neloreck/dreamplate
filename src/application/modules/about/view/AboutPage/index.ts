/**
 * @module application/about
 */

export * from "./AboutPage.component";
