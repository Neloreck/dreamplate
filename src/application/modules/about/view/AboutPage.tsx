import { useManager } from "dreamstate";
import { ReactElement } from "react";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";

import { RouterManager } from "@/core/data/store";
import { MainHeader } from "@/core/view/components";
import { IApplicationTheme } from "@/lib/theme";

export const useStyles = createUseStyles(({ spacing }: IApplicationTheme) => ({
  content: {
    "& > button": {
      margin: spacing.unit * 2
    },
    alignItems: "center",
    alignSelf: "normal",
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    overflowY: "auto",
    padding: spacing.unit * 2
  }
}));

export function AboutPage({
  classes: { content } = useStyles(),
  routerContext: { history } = useManager(RouterManager)
}): ReactElement {
  return (
    <>
      <MainHeader />

      <main className={content}>
        About page.
        <Link to={history.createHref("/home")} reloadDocument>
          <button>Go home</button>
        </Link>
      </main>
    </>
  );
}
