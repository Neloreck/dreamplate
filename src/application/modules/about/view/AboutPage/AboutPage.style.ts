import { createUseStyles } from "react-jss";

// Lib.
import { IApplicationTheme } from "@Lib/theme";

export const useStyles = createUseStyles(({ palette }: IApplicationTheme) => ({
  content: {
    alignItems: "center",
    alignSelf: "normal",
    backgroundSize: "cover",
    display: "flex",
    flexDirection: "column",
    flexGrow: 1,
    justifyContent: "center",
    overflowY: "auto"
  }
}));
