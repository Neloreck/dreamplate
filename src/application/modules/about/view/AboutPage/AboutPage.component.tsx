import { useManager } from "dreamstate";
import { ReactElement, useCallback } from "react";

import { RouterContextManager } from "@Core/data/store";
import { MainHeader } from "@Core/view/components/MainHeader";

import { useStyles } from "./AboutPage.style";

import "@Lib/components/custom/CustomButton";

export function AboutPage({
  classes: { content } = useStyles(),
  routerContext: { routingActions: { hardPush } } = useManager(RouterContextManager)
}): ReactElement {
  const onHomeNavigated = useCallback(() => hardPush("/home"), []);

  return (
    <>

      <MainHeader/>

      <main className={content}>

        About page.

        <custom-button onClick={onHomeNavigated}>
          Go home
        </custom-button>

      </main>

    </>
  );
}
