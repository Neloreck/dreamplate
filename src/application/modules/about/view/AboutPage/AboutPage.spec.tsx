import "@Test/utils/mock_custom_elements";

// Test.
import { getThemedComponent, withMockedTheme } from "@Lib/theme/testing";
import { shallow } from "enzyme";

// View.
import { AboutPage, IAboutPageInjectedProps } from "./AboutPage.component";

describe("About page component rendering.", () => {

  it("Should match HOC snapshot.", () => {

    const wrapper = shallow(<AboutPage {...{} as IAboutPageInjectedProps}/>);

    expect(wrapper).toMatchSnapshot("Default HOC layout.");
  });

  it("Should match mounted snapshot with default theme.", () => {

    const wrapper = shallow(withMockedTheme(<AboutPage {...{} as IAboutPageInjectedProps}/>));
    const mounted = getThemedComponent(wrapper);

    expect(mounted).toMatchSnapshot("Mounted with default theme.");
  });
});
