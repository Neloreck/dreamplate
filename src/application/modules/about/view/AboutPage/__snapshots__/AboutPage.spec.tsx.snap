// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About page component rendering. Should match HOC snapshot.: Default HOC layout. 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`About page component rendering. Should match mounted snapshot with default theme.: Mounted with default theme. 1`] = `
<AboutPage
  classes={
    Object {
      "content": "AboutPage-content-0-2-1",
    }
  }
>
  <JssContextSubscriber>
    <WithStyles(MainHeader)
      innerRef={null}
      jssContext={
        Object {
          "classNamePrefix": "",
          "disableStylesGeneration": false,
        }
      }
      theme={
        Object {
          "palette": Object {
            "background": Object {
              "default": "#FFF",
              "paper": "#eeeeee",
            },
            "primary": Object {
              "dark": "#5C6BC0",
              "light": "#9fa8da",
              "main": "#7986cb",
            },
            "secondary": Object {
              "dark": "#ab47bc",
              "light": "#7986cb",
              "main": "#ba68c8",
            },
            "text": Object {
              "primary": "#000",
              "secondary": "#444444",
            },
            "type": "light",
          },
          "spacing": Object {
            "unit": 4,
          },
        }
      }
    >
      <MainHeader
        classes={
          Object {
            "root": "MainHeader-root-0-2-2",
          }
        }
      >
        <header
          className="MainHeader-root-0-2-2"
        >
          <custom-heading
            size={3}
            text="Boilerplate"
          />
          <custom-button
            onClick={[Function]}
          >
            Toggle
          </custom-button>
        </header>
      </MainHeader>
    </WithStyles(MainHeader)>
  </JssContextSubscriber>
  <main
    className="AboutPage-content-0-2-1"
  >
    About page.
    <custom-button
      onClick={[Function]}
    >
      Go home
    </custom-button>
  </main>
</AboutPage>
`;
